<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <!-- <arg name="gui" default="true"/>
    <include file="$(find nti_autonet_2021_car_sim)/launch/world1.launch" />
    <include file="$(find nti_autonet_2021_car_sim)/launch/spawn.launch" /> -->

    <param name="robot_description" command="$(find xacro)/xacro --inorder '$(find nti_autonet_2021_car_sim)/urdf/vehicle_v1_double.xacro'" />
    
    <rosparam file="$(find nti_autonet_2021_car_sim)/config/control.yaml" command="load" />

    <!-- load the controllers -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/vehicle_v1" args="front_left_rev_controller front_right_rev_controller 
    front_left_cont_controller front_right_cont_controller joint_state_controller" />
    <!-- <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/vehicle_v1" args="mobile_base_controller joint_state_controller" /> -->
    
    <arg name="robot" default="machines"/>
    <arg name="debug" default="false"/>
    <arg name="gui" default="true"/>
    <arg name="headless" default="false"/>
    <arg name="pause" default="false"/>
    <arg name="world" default="map1" />
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find nti_autonet_2021_car_sim)/worlds/$(arg world).world"/>
        <arg name="debug" value="$(arg debug)" />
        <arg name="gui" value="$(arg gui)" />
        <arg name="paused" value="$(arg pause)"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="headless" value="$(arg headless)"/>
        <env name="GAZEBO_MODEL_PATH" value="$(find nti_autonet_2021_car_sim)/models:$(optenv GAZEBO_MODEL_PATH)"/>
    </include>


    <arg name="x" default="-2.240799" />
    <arg name="y" default="-2" />
    <arg name="z" default="0.2" />
    <!-- <arg name="R" default="-2.240799" />
    <arg name="P" default="-2" /> -->
    <arg name="Y" default="-1.345563" />

    
    <!--0 -0.1 0.2" rpy="0 0.5 0  -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
    
    <node name="mybot_spawn" pkg="gazebo_ros" type="spawn_model" output="screen" args="-urdf -param robot_description -model vehicle_v1 -x $(arg x) -y $(arg y) -z $(arg z) -Y $(arg Y)" />
    <!-- <node name="twist_to_ackermann" pkg="nti_autonet_2021_car_sim" type="twist_to_ackermann" output="screen">
    </node> -->
</launch>
